ember-cli/2017-03/march-31.md

```diff
@@ -0,0 +1,127 @@
+# ember-cli Meeting Notes - 2017-03-31
+
+## Attendees
+
+- Nathan Hammond
+- Jake Bixby
+- Tobias Bieniek
+- David Hamilton
+- Alex Navasardyan
+- Katie Gengler
+- Kelly Selden
+- Chad Hietala
+- Robert Jackson
+
+## Agenda
+
+- Glimmer Pipeline
+- Tree Shaking
+- `app.import()` from `node_modules`
+- Blueprint (`blprnt`)
+  - use Glimmer resolver?
+  - composability of blueprints
+  - blueprint migrations/transactional?
+- Background Service
+- Slow Commands
+  - `generate`
+  - `help`
+- Ember CLI/Broccoli Errors
+- Global add-ons
+- `lib-cli`
+
+## Discussion Notes
+
+### Glimmer Pipeline
+
+Extract `lib/broccoli/ember-app.js` into a separate npm package.
+
+### Tree Shaking
+
+- given list of file entries + module information + roots, generate a retained graph
+- keep module loose w/o concat until the very end
+- generate dep graph at the very end of the build process
+- have a hook that will give you access to dep graph w/o being able to modify it
+
+### `app.import` from `node_modules`
+
+- support amd, cjs or es6 w/o watching
+- everything that's imported is considered dynamically linked
+- should care about single file (entry point) make sure we can tell the difference between globals/modules for tree shaking (globals and module are in different buckets)
+- need to be able to "blacklist" modules to resolve them during runtime
+  - possibly make it automatic, we *should* have info already or do a dep walk on app.import path
+  - should *not* require for users to pass into list of modules when using `app.import`
+- address differences between `import`s (documentation or otherwise)
+  - `app.import` is project relative
+  - `this.import` is relative to whatever `this` is
+- path *has* to start with `vendor`, `node_modules`
+
+### Blrpnt extraction
+
+- unfork
```

Comment:
The biggest work item here, is ember-cli blueprint test helpers being a-bit funky. The blrpnt effort would actually make writing that test helper much simpler (no longer needs ember-cli itself etc...)

If someone wants to work on this, please feel free to reach out.